internal/handlers/commonpath/path_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commonpath
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+func TestGetPathValue(t *testing.T) {
+	t.Parallel()
+
+	emptyDoc := new(types.Document)
+	docDoc := must.NotFail(types.NewDocument("foo", must.NotFail(types.NewDocument("bar", 1))))
+	docArrayOne := must.NotFail(types.NewDocument("foo", must.NotFail(types.NewArray(
+		must.NotFail(types.NewDocument("bar", 1)),
+	))))
+	docArrayTwo := must.NotFail(types.NewDocument("foo", must.NotFail(types.NewArray(
+		must.NotFail(types.NewDocument("bar", 1)),
+		must.NotFail(types.NewDocument("bar", 2)),
+	))))
```

Comment:
So, we only consider cases when the array contains a document?

For example, a case where `doc` is `{"foo": [1, 2, 3]}` and `path` is `foo.1` is not included (and as far as I can see will return an empty result). Should we highlight that `FindValues` only finds array's elements that are documents but not scalars?

Reply:
It looks to me that the previous code was able to work with scalars, see this line: https://github.com/FerretDB/FerretDB/pull/3087/files#diff-3925922352e709447e9f2cf1c69b321f762441108abe68c4ffacded239bc8c40L315. But maybe I misunderstand something.